tests/test_commands.py

```
@@ -1197,8 +1247,80 @@ async def test_update_firmware_error_response(test_charger, mock_aioclient):
+    }
+
+    mock_aioclient.get(
+        "https://api.github.com/repos/OpenEVSE/ESP32_WiFi_V4.x/releases/latest",
+        status=200,
+        body=json.dumps(github_response),
+    )
+
+    with caplog.at_level(logging.DEBUG):
+        with pytest.raises(
+            RuntimeError,
+            match="Could not resolve latest firmware download URL from GitHub.",
```

Ruff (0.15.15)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)